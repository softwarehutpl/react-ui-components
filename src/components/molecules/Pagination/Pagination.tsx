import React from 'react';

export const Pagination: React.StatelessComponent<{}> = () => (
  <div className="">Pagination
  </div>
);
