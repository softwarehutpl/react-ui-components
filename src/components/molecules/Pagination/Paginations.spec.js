import React from 'react';
import { create, act } from 'react-test-renderer';
import 'jest-styled-components';
import Pagination, { PaginationWrapper, IconComponent, PageNumber, PaginationItem } from './Pagination';

describe('Pagination component', () => {
  it('should render corectly', () => {
    const component = create(<Pagination />);
    expect(component).toMatchSnapshot();
  });

  it('should render corectly', () => {
    const component = create(<PaginationWrapper />);
    expect(component).toMatchSnapshot();
  });

  it('should render corectly', () => {
    const component = create(<IconComponent />);
    expect(component).toMatchSnapshot();
  });

  it('should render corectly', () => {
    const component = create(<PageNumber />);
    expect(component).toMatchSnapshot();
  });

  it('should render corectly', () => {
    const component = create(<PaginationItem />);
    expect(component).toMatchSnapshot();
  });

  it('should render PaginationWrapper component with default style', () => {
    const component = create(<PaginationWrapper />);
    const paginationWrapper = component.toJSON();
    expect(paginationWrapper).toHaveStyleRule('width', '100%');
    expect(paginationWrapper).toHaveStyleRule('height', 'auto');
    expect(paginationWrapper).toHaveStyleRule('background-color', 'transparent');
    expect(paginationWrapper).toHaveStyleRule('display', 'flex');
    expect(paginationWrapper).toHaveStyleRule('justify-content', 'center');
    expect(paginationWrapper).toHaveStyleRule('align-items', 'center');
    expect(paginationWrapper).toHaveStyleRule('flex-direction', 'column');
  });

  it('should render PaginationItem component with default style', () => {
    const component = create(<PaginationItem />);
    const paginationWrapper = component.toJSON();
    expect(paginationWrapper).toHaveStyleRule('width', '300px');
    expect(paginationWrapper).toHaveStyleRule('border-radius', '0px');
    expect(paginationWrapper).toHaveStyleRule('border', 'none');
    expect(paginationWrapper).toHaveStyleRule('box-shadow', '0px 3px 9px rgba(100,100,100,0.3)')
  });

  it('should render PaginationWrapper component with provided style', () => {
    const component = create(<PaginationWrapper
      width={900}
      height={900}
      wrapperBackgroundColor="#333"
    />);
    const paginationWrapper = component.toJSON();
    expect(paginationWrapper).toHaveStyleRule('width', '900px');
    expect(paginationWrapper).toHaveStyleRule('height', '900px');
    expect(paginationWrapper).toHaveStyleRule('background-color', '#333');
  });

  it('should render PaginationItem component with provided style', () => {
    const component = create(<PaginationItem
      itemWidth={400}
      borderRadius={10}
      borderColor="#444"
    />);
    const paginationWrapper = component.toJSON();
    expect(paginationWrapper).toHaveStyleRule('width', '400px');
    expect(paginationWrapper).toHaveStyleRule('border-radius', '10px');
    expect(paginationWrapper).toHaveStyleRule('border', '1px solid #444');
  });

  it('should call add to cart with onClick method', () => {
    const mockOnClick = jest.fn();
    const component = create(<IconComponent onClick={mockOnClick} />);
    const instance = component.root;
    const button = instance.findByType('button');
    act(() => {
      button.props.onClick();
    });
    expect(mockOnClick.mock.calls.length).toEqual(1);
  });

  it('should call handle change page method', () => {
    const mockOnClick = jest.fn();
    const component = create(<PageNumber onClick={mockOnClick} />);
    const instance = component.root;
    const button = instance.findByType('button');
    act(() => {
      button.props.onClick();
    });
    expect(mockOnClick.mock.calls.length).toEqual(1);
  });
});
