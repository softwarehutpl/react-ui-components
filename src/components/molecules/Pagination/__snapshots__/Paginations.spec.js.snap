// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should render corectly 1`] = `
.c0 {
  width: 100%;
  height: auto;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 1100px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 50px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
</div>
`;

exports[`Pagination component should render corectly 2`] = `
.c0 {
  width: 100%;
  height: auto;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`Pagination component should render corectly 3`] = `
.c0 {
  position: absolute;
  top: 10px;
  right: 10px;
  z-index: 2;
  width: 30px;
  height: 30px;
  background-color: transparent;
  cursor: pointer;
  border: none;
}

.c0:focus {
  outline: none;
}

<button
  className="c0"
/>
`;

exports[`Pagination component should render corectly 4`] = `
.c0 {
  margin: 5px;
  padding: 10px;
  cursor: pointer;
  font-weight: 400;
  border: none;
  background-color: transparent;
}

.c0:focus {
  outline: none;
}

<button
  className="c0"
/>
`;

exports[`Pagination component should render corectly 5`] = `
.c0 {
  width: 300px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0px 3px 9px rgba(100,100,100,0.3);
  border-radius: 0px;
  border: none;
  margin-bottom: 50px;
  position: relative;
}

<div
  className="c0"
/>
`;
